test/e2e/run_staticip_test.go

```diff
@@ -105,6 +105,11 @@ var _ = Describe("Podman run with --ip flag", func() {
 		result.WaitWithDefaultTimeout()
 		Expect(result).Should(Exit(0))
 
+		// We need to set "no_proxy" in proxy environment
+		env := os.Getenv("no_proxy")
+		os.Setenv("no_proxy", ip)
+		defer os.Setenv("no_proxy", env)
```

Comment:
This will set no_proxy even when it was unset before, use LookupEnv and only set the env when it returned true.